src/Servant/Swagger/Internal.hs

```diff
@@ -146,70 +146,86 @@ addDefaultResponse400 pname = setResponseWith (\old _new -> alter400 old) 400 (r
+instance (ToSchema a, AllAccept cs, AllToResponseHeader hs, KnownNat c) => HasSwagger (Verb 'GET c cs (Headers hs a)) where
+  toSwagger = mkEndpoint "/" get (fromIntegral (natVal (Proxy :: Proxy c)))
 
-instance AllAccept cs => HasSwagger (Get cs ()) where
+instance AllAccept cs => HasSwagger (Verb 'GET c cs ()) where
   toSwagger = noContentEndpoint "/" get
 
 -- -----------------------------------------------------------------------
 -- PATCH
 -- -----------------------------------------------------------------------
 
-instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Patch cs a) where
+instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Verb 'PATCH c cs a) where
   toSwagger _ = toSwagger (Proxy :: Proxy (Patch cs (Headers '[] a)))
 
-instance (ToSchema a, AllAccept cs, AllToResponseHeader hs) => HasSwagger (Patch cs (Headers hs a)) where
-  toSwagger = mkEndpoint "/" patch 200
+instance (ToSchema a, AllAccept cs, AllToResponseHeader hs, KnownNat c) => HasSwagger (Verb 'PATCH c cs (Headers hs a)) where
+  toSwagger = mkEndpoint "/" patch (fromIntegral (natVal (Proxy :: Proxy c)))
 
-instance AllAccept cs => HasSwagger (Patch cs ()) where
+instance AllAccept cs => HasSwagger (Verb 'PATCH c cs ()) where
   toSwagger = noContentEndpoint "/" patch
 
 -- -----------------------------------------------------------------------
 -- PUT
 -- -----------------------------------------------------------------------
 
-instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Put cs a) where
+instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Verb 'PUT c cs a) where
   toSwagger _ = toSwagger (Proxy :: Proxy (Put cs (Headers '[] a)))
 
-instance (ToSchema a, AllAccept cs, AllToResponseHeader hs) => HasSwagger (Put cs (Headers hs a)) where
-  toSwagger = mkEndpoint "/" put 200
+instance (ToSchema a, AllAccept cs, AllToResponseHeader hs, KnownNat c) => HasSwagger (Verb 'PUT c cs (Headers hs a)) where
+  toSwagger = mkEndpoint "/" put (fromIntegral (natVal (Proxy :: Proxy c)))
 
-instance AllAccept cs => HasSwagger (Put cs ()) where
+instance AllAccept cs => HasSwagger (Verb 'PUT c cs ()) where
   toSwagger = noContentEndpoint "/" put
 
 -- -----------------------------------------------------------------------
 -- POST
 -- -----------------------------------------------------------------------
 
-instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Post cs a) where
+instance OVERLAPPABLE_ (ToSchema a, AllAccept cs) => HasSwagger (Verb 'POST c cs a) where
   toSwagger _ = toSwagger (Proxy :: Proxy (Post cs (Headers '[] a)))
 
-instance (ToSchema a, AllAccept cs, AllToResponseHeader hs) => HasSwagger (Post cs (Headers hs a)) where
-  toSwagger = mkEndpoint "/" post 201
+instance (ToSchema a, AllAccept cs, AllToResponseHeader hs, KnownNat c) => HasSwagger (Verb 'POST c cs (Headers hs a)) where
+  toSwagger = mkEndpoint "/" post (fromIntegral (natVal (Proxy :: Proxy c)))
```

Comment:
Wouldn't it be much easier to read if it wasn't called `c` but e.g. `status`? Especially with the `cs` afterwards.